### PR TITLE
SearchEngineTracker: Re-enable DefaultSearchEngineP3A test.

### PR DESCRIPTION
**Pull request by rillian**
This was disabled in 27bf9e3b0b97 for intermittent failures.
Add a call to wait for TemplateURLServiceFactory to finish
loading, and re-enable to see if that stabilizes the browser
test.

See https://github.com/brave/brave-browser/issues/13057 for tracking.

This duplicates VerifyTemplateURLServiceLoad() from the
search_engine_provider browser test, but it's a small piece
of code and having two file-local implementations is simple
enough.

Resolves 

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: browser/search_engines/search_engine_tracker_browsertest.cc [hunk 1]
```diff
@@ -26,14 +27,28 @@ class SearchEngineProviderP3ATest : public InProcessBrowserTest {
...
   histogram_tester_->ExpectUniqueSample(kDefaultSearchEngineMetric,
                                         SearchEngineP3A::kGoogle, 1);
 
-  auto* service = TemplateURLServiceFactory::GetForProfile(
-      browser()->profile());
+  auto* service =
+      TemplateURLServiceFactory::GetForProfile(browser()->profile());
+
+  // Make sure the service is initialized.
+  EXPECT_TRUE(VerifyTemplateURLServiceLoad(service));
```
Comment (iefremov) [inline browser/search_engines/search_engine_tracker_browsertest.cc:51]: frankly speaking I don't quite understand why simple `search_test_utils::WaitForTemplateURLServiceToLoad(service);` is not enough. It already asserts that the service is loaded inside, so this wrapper looks really redundant

Comment (rillian) [reply browser/search_engines/search_engine_tracker_browsertest.cc]: Seems fine to me too! I've made it so.

**Comment by rillian** [comment]
As far as I can tell, the intermittent macOS ci failures aren't related.

**Comment by bsclifton** [comment]
Nice work 😄 Thanks for getting these back in @rillian!

**Pull request merged**